tools/quickstart/quickstart/docker.py

```diff
@@ -32,29 +64,43 @@ def update_and_start(host_base_dir: str) -> None:
         )
 
     main_docker_service_names = ["trustlines-node", "watchtower"]
-    optional_docker_service_names = get_optional_docker_service_names()
+    optional_docker_service_names = get_optional_docker_service_names(base_dir)
     all_docker_service_names = (
         main_docker_service_names + optional_docker_service_names + ["tlbc-monitor"]
     )
 
-    base_docker_environment_variables = {**os.environ, "HOST_BASE_DIR": host_base_dir}
-    if is_validator_account_prepared():
-        docker_environment_variables = {
-            **base_docker_environment_variables,
-            "VALIDATOR_ADDRESS": get_validator_address(),
+    default_env_vars = {"COMPOSE_PROJECT_NAME": project_name}
+    if is_validator_account_prepared(base_dir):
+        env_variables = {
+            **default_env_vars,
+            "VALIDATOR_ADDRESS": get_validator_address(base_dir),
             "ROLE": "validator",
         }
         click.echo("\nNode will run as a validator")
     else:
-        docker_environment_variables = {
-            **base_docker_environment_variables,
+        env_variables = {
+            **default_env_vars,
             "VALIDATOR_ADDRESS": "",
             "ROLE": "observer",
         }
         click.echo("\nNode will run as a non-validator")
 
+    with open(os.path.join(base_dir, ".env"), mode="w") as env_file:
+        env_file.writelines(
+            f"{key}={value}\n" for (key, value) in env_variables.items()
+        )
+
+    runtime_env_variables = {
+        **os.environ,
+        **env_variables,
```

Comment:
Any reason why we `env_variables` are both written to `env_file` and used as runtime vars?

Reply:
Yes, the file is for later running it with the docker compose file, the runtime vars for the setup during the quickstart. Because we are writing it to a different folder, the env file is not so easy accessible during quickstart, so I decided the easiest is just to add it to runtime as well. The other option would be to switch to the right folder before executing the docker compose commands, do you think that is better?

Reply:
I see, makes sense. I'm ok with the current solution then.